Java/core/src/main/java/com/amazon/randomcutforest/store/NodeStore.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.store;
+
+import com.amazon.randomcutforest.tree.BoundingBox;
+
+/**
+ * A fixed-size buffer for storing interior tree nodes. An interior node is
+ * defined by its location in the tree (parent and child nodes), its random cut,
+ * and its bounding box. The NodeStore class uses arrays to store these field
+ * values for a collection of nodes. An index in the store can be used to look
+ * up the field values for a particular node.
+ *
+ * If we think of an array of Node objects as being row-oriented (where each row
+ * is a Node), then this class is analogous to a column-oriented database of
+ * Nodes.
+ *
+ * Note that a NodeStore does not store instances of the
+ * {@link com.amazon.randomcutforest.tree.Node} class.
+ */
+public class NodeStore extends SmallStoreCommon {
+
+    public final short[] parentIndex;
+    public final short[] leftIndex;
+    public final short[] rightIndex;
+    public final int[] cutDimension;
+    public final float[] cutValue;
+    public final int[] mass;
```

Comment:
question. does mass need int?

Reply:
It probably doesn't, but the mass value is already defined as an int in the existing Node class and I thought we should keep the int here for consistency. I think it would make sense to change both to short, but maybe in a separate PR since this one is already pretty big.

Reply:
ok. this be a TODO.

Reply:
mass can be a short (sample size of a tree)

Reply:
mass can be short.